modules/data/src/main/java/com/opengamma/strata/data/scenario/CurrencyScenarioArray.java

```diff
@@ -142,7 +142,64 @@ public CurrencyScenarioArray convertedTo(Currency reportingCurrency, ScenarioFxR
         amounts.getValues().mapWithIndex((i, v) -> v * fxRateProvider.fxRate(getCurrency(), reportingCurrency, i));
     return of(reportingCurrency, convertedValues);
   }
-
+  
+  //-------------------------------------------------------------------------
+  /**
+   * Returns a new array containing the values from this array added to the values in the other array.
+   * <p>
+   * The amounts are added to the matching element in this array.
+   * The currency must be the same as the currency of this array.
+   * The arrays must have the same size.
+   *
+   * @param other  another array of multiple currency values.
+   * @return a new array containing the values from this array added to the values in the other array
+   * @throws IllegalArgumentException if the arrays have different sizes
+   */
+  public CurrencyScenarioArray plus(CurrencyScenarioArray other) {
+    return CurrencyScenarioArray.of(amounts.plus(other.amounts));
+  }
+  
+  /**
+   * Returns a new array containing the values from this array with the values from the amount added.
+   * <p>
+   * The amount is added to each element in this array.
+   * The currency must be the same as the currency of this array.
+   *
+   * @param amount  the amount to add
+   * @return a new array containing the values from this array added to the values in the other array
+   */
+  public CurrencyScenarioArray plus(CurrencyAmount amount) {
+    return CurrencyScenarioArray.of(amounts.plus(amount));
+  }
+  
+  /**
+   * Returns a new array containing the values from this array with the values from the other array subtracted.
+   * <p>
+   * The amounts are subtracted from the matching element in this array.
+   * The currency must be the same as the currency of this array.
+   * The arrays must have the same size.
+   *
+   * @param other  another array of multiple currency values.
+   * @return a new array containing the values from this array added with the values from the other array subtracted
```

Comment:
Not "added"

Reply:
Confession time; I copy/pasted all of these new Javadoc from CurrencyAmountArray!

Will clean that up as well.